pkg/init/backend/interactive.go

```diff
@@ -34,6 +36,18 @@ func (o *InteractiveBackend) Validate(flags map[string]string) error {
 	return nil
 }
 
+// DetectFramework uses the anlizer library in order to detect the language
+// as well as framework and tool of a Devfile location
+// Passes in PATH and gets a function that contains strings describing the
+// directory's detected languages.
+func DetectFramework(path string) ([]language.Language, error) {
```

Comment:
You should instead create a new AlizerBackend, and place this code in its implementation of `SelectDevfile`. You can look at https://github.com/redhat-developer/odo/pull/5434 for more details on how the `pkg/init` package has been designed